// ligdicash/index.js


// Copyright 9-12-2020 (c) Louis Bertson <louisbertson@gmail.com>
//
// Permission is hereby granted, free of charge, to any person obtaining a
// copy of this software and associated documentation files (the
// "Software"), to deal in the Software without restriction, including
// without limitation the rights to use, copy, modify, merge, publish,
// distribute, sublicense, and/or sell copies of the Software, and to permit
// persons to whom the Software is furnished to do so, subject to the
// following conditions:
//
// The above copyright notice and this permission notice shall be included
// in all copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE\
// USE OR OTHER DEALINGS IN THE SOFTWARE.

const axios = require('axios');

var HEADERS = {};
var commande = {};
commande.invoice = {};
commande.invoice.items = [];
commande.invoice.devise = 'xof';
commande.invoice.description = '';
commande.invoice.total_amount = '';
commande.invoice.otp = '';
commande.store = {};
commande.store.name = 'AGRAVI BURKINA';
commande.store.website_url = 'https//www.agraviburkina.com';
commande.actions = {};
commande.actions.return_url = '';
commande.actions.cancel_url = '';
commande.actions.callback_url = '';
commande.custom_data = {};
commande.custom_data.boss = 'Arnaud Salembere';
commande.custom_data.lieu = 'Kalgodin';
var self;
function LigdiCash(returnUrl,cancelUrl,callbackUrl) {
	 self =this;

	// commande.actions.returnUrl = returnUrl;
	commande.actions.return_url = returnUrl;
	commande.actions.cancel_url = cancelUrl;
	commande.actions.callback_url = callbackUrl;
	this.url = 'https://app.ligdicash.com/pay/v01/redirect/checkout-invoice/create';
};
function  load_items(order,command,controller, callback){
	command.invoice.description = order.note;
		command.invoice.customer = '';
		command.invoice.customer_firstname = order.firstname;
		command.invoice.customer_lastname = order.lastname;
		command.invoice.email = order.email;
	   command.invoice.total_amount = order.price;
	   for(var i = 0; i<order.items.length;i++){
		   var item = {};
		   item.name = order.items[i].name;
		   item.description = order.items[i].reference;
		   item.quantity = order.items[i].count;
		   item.unit_price = order.items[i].price;
		   item.total_price = order.items[i].count * order.items[i].price;
		   command.invoice.items.push(item);
	   }
	callback(command);
};
LigdiCash.prototype.post = function(order, controller,callback) {
	   load_items(order,commande,controller,function(cmd) {
		   var donnee = {};
	   donnee.commande = cmd;
	   var data = JSON.stringify(donnee);
	   data = data.parseJSON(true);
	var headers = {};
	axios.defaults.headers.common['User-Agent'] = 'LigdiCashpaymentCheckoutforNode.js';
	axios.defaults.headers.common['Accept'] = 'application/json';
	axios.defaults.headers.common['Content-Type'] = 'application/json';
	//axios.defaults.headers.common['Apikey'] = 'ZVBGJYG9VS9B283ES';
	axios.defaults.headers.common['Apikey'] = 'NK94SH1QBR1WEGBLR';
	axios.defaults.headers.common['Authorization'] = 'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZF9hcHAiOiIxODAzIiwiaWRfYWJvbm5lIjo2MzQyNiwiZGF0ZWNyZWF0aW9uX2FwcCI6IjIwMjItMDItMTggMTk6MDg6MDIifQ.PFnNuNTfqGdLC3wRlZYMmQx99I50jUg5xhiOPS7Lf80';
	//axios.defaults.headers.common['Authorization'] = 'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZF9hcHAiOiI0MjQiLCJpZF9hYm9ubmUiOiI2MzQyNiIsImRhdGVjcmVhdGlvbl9hcHAiOiIyMDIwLTEyLTEyIDEyOjE5OjUxIn0.7BkQviVdzbhgSEvA2J-Kog1fKGKaO1_lcfhZ1yAA8Eg';
		console.log(data);

		axios.post(self.url,data,{})
	   .then(function(res){
		   callback({data : res.data,id : controller.id});
	   });
	});
	return self;


};
LigdiCash.prototype.detail = function(token,callback){
	var verifyUrl = 'https://app.ligdicash.com/pay/v01/redirect/checkout-invoice/confirm?invoiceToken=';

   if(!token){
	callback(null, { type : 'error', message : 'Token is missing'});
   }
	axios.defaults.headers.common['User-Agent'] = 'LigdiCashpaymentCheckoutforNode.js';
	axios.defaults.headers.common['Accept'] = 'application/json';
	axios.defaults.headers.common['Content-Type'] = 'application/json';
	//axios.defaults.headers.common['Apikey'] = 'ZVBGJYG9VS9B283ES';
	axios.defaults.headers.common['Apikey'] = 'NK94SH1QBR1WEGBLR';
	axios.defaults.headers.common['Authorization'] = 'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZF9hcHAiOiIxODAzIiwiaWRfYWJvbm5lIjo2MzQyNiwiZGF0ZWNyZWF0aW9uX2FwcCI6IjIwMjItMDItMTggMTk6MDg6MDIifQ.PFnNuNTfqGdLC3wRlZYMmQx99I50jUg5xhiOPS7Lf80';
	//axios.defaults.headers.common['Authorization'] = 'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZF9hcHAiOiI0MjQiLCJpZF9hYm9ubmUiOiI2MzQyNiIsImRhdGVjcmVhdGlvbl9hcHAiOiIyMDIwLTEyLTEyIDEyOjE5OjUxIn0.7BkQviVdzbhgSEvA2J-Kog1fKGKaO1_lcfhZ1yAA8Eg';
	axios.get(verifyUrl+token)
	   .then(function(res){
		console.log(res);
		   callback({data : res.data, token : token});
	   });
};


exports.LigdiCash = LigdiCash;
exports.init = function(returnUrl,cancelUrl,callbackUrl) {
	return new LigdiCash(returnUrl,cancelUrl,callbackUrl);
};
exports.create = function(returnUrl,cancelUrl,callbackUrl) {
	return exports.init(returnUrl,cancelUrl,callbackUrl);
};
